invoke/tasks.py

```diff
@@ -215,8 +215,9 @@ def arg_opts(self, name, default, taken_names):
                 opts["kind"] = kind
             opts["default"] = default
         # Help
-        if name in self.help:
-            opts["help"] = self.help[name]
+        help_name_key = name if name in self.help else opts.get("attr_name")
+        if help_name_key in self.help:
+            opts["help"] = self.help.pop(help_name_key)
```

Comment:
On line 61 above, we could perhaps use `self.help = help.copy() if help else {}` rather than help directly, in case someone is passing the same help dictionary to multiple decorators to avoid copy-pasting

Reply:
About to read all this over myself, but I think I agree; that sort of defensive/functional-style programming is something I am frequently worrying about but not always finding the time to actually implement 🙃